core/src/test/java/org/apache/iceberg/rest/requests/TestReportMetricsRequestParser.java

```diff
@@ -0,0 +1,127 @@
+
+    Assertions.assertThatThrownBy(
+            () -> ReportMetricsRequestParser.fromJson("{\"report-type\":\"scan-report\"}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing string: table-name");
+
+    Assertions.assertThatThrownBy(
+            () ->
+                ReportMetricsRequestParser.fromJson(
+                    "{\"report-type\":\"scan-report\", \"table-name\" : \"x\"}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing long: snapshot-id");
+  }
+
+  @Test
+  public void invalidReportType() {
+    Assertions.assertThatThrownBy(
+            () -> ReportMetricsRequestParser.fromJson("{\"report-type\":\"invalid\"}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage(
+            "No enum constant org.apache.iceberg.rest.requests.ReportMetricsRequest.ReportType.INVALID");
+
+    Assertions.assertThatThrownBy(
+            () ->
+                ReportMetricsRequestParser.fromJson(
+                    ReportMetricsRequestParser.toJson(
+                        ReportMetricsRequest.of(new MetricsReport() {}))))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage(
+            "Unsupported report type: org.apache.iceberg.rest.requests.TestReportMetricsRequestParser$1");
+  }
+
+  @Test
+  public void roundTripSerde() {
+    String tableName = "roundTripTableName";
+    Schema projection =
+        new Schema(Types.NestedField.required(1, "c1", Types.StringType.get(), "c1"));
+    ScanReport scanReport =
+        ImmutableScanReport.builder()
+            .tableName(tableName)
+            .projection(projection)
+            .snapshotId(23L)
+            .filter(Expressions.alwaysTrue())
+            .scanMetrics(ScanMetricsResult.fromScanMetrics(ScanMetrics.noop()))
+            .build();
+
+    String expectedJson =
+        "{\n"
+            + "  \"report-type\" : \"scan-report\",\n"
+            + "  \"table-name\" : \"roundTripTableName\",\n"
+            + "  \"snapshot-id\" : 23,\n"
+            + "  \"filter\" : true,\n"
+            + "  \"projection\" : {\n"
+            + "    \"type\" : \"struct\",\n"
+            + "    \"schema-id\" : 0,\n"
+            + "    \"fields\" : [ {\n"
+            + "      \"id\" : 1,\n"
+            + "      \"name\" : \"c1\",\n"
+            + "      \"required\" : true,\n"
+            + "      \"type\" : \"string\",\n"
+            + "      \"doc\" : \"c1\"\n"
+            + "    } ]\n"
+            + "  },\n"
+            + "  \"metrics\" : { }\n"
+            + "}";
+
+    ReportMetricsRequest metricsRequest = ReportMetricsRequest.of(scanReport);
+
+    String json = ReportMetricsRequestParser.toJson(metricsRequest, true);
+    Assertions.assertThat(json).isEqualTo(expectedJson);
+
+    Assertions.assertThat(ReportMetricsRequestParser.fromJson(json).report())
+        .usingRecursiveComparison()
+        .ignoringFields("projection")
```

Comment:
Why is this needed? It seems strange to me that the projection doesn't match.

Reply:
there is no `equals()` method defined on `Schema`, so it will always perform reference equality check